dev/ci/bazel-prechecks.sh

```
@@ -1,20 +1,15 @@
 #!/usr/bin/env bash
 
 set -eu
+EXIT_CODE=0
 
 bazelrc="--bazelrc=.bazelrc --bazelrc=.aspect/bazelrc/ci.bazelrc --bazelrc=.aspect/bazelrc/ci.sourcegraph.bazelrc"
 
-# We run :gazelle since currently `bazel configure` tries to execute something with go and it doesn't exist on the bazel agent
 echo "--- :bazel: Running bazel configure"
-bazel "${bazelrc}" configure
-
-# We disable exit on error here, since we want to catch the exit code and interpret it
-set +e
+bazel $bazelrc configure
 
 echo "--- Checking if BUILD.bazel files were updated"
-git diff --exit-code
-
-EXIT_CODE=$?
+git diff --exit-code || EXIT_CODE=$? # do not fail on non-zero exit
```

A couple of unrelated pieces:
1) quoting `"${bazelrc}"` meant we were passing all the RC params as a single arg. This caused the command to fail.
2) We ran `set +e`, so the failure was being ignored, so this whole CI step wasn't doing anything. See [this](https://buildkite.com/sourcegraph/sourcegraph/builds/238767#0189fecf-96f8-478f-8f0b-171a13c5125a) build for an example.
3) This makes it so we just capture the exit code of the `git diff` command without unsetting "fail on error" globally.